modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -69,21 +69,30 @@ export default class MVTLayer extends TileLayer {
 
     if (tileJSON) {
       data = tileJSON.tiles;
+    }
+
+    this.setState({data, tileJSON});
+  }
+
+  _getTilesetOptions(props) {
+    const opts = super._getTilesetOptions(props);
+    const {tileJSON} = this.state;
 
-      if (Number.isFinite(tileJSON.minzoom) && tileJSON.minzoom > minZoom) {
-        minZoom = tileJSON.minzoom;
+    if (tileJSON) {
+      if (Number.isFinite(tileJSON.minzoom) && tileJSON.minzoom > props.minZoom) {
+        opts.minZoom = tileJSON.minzoom;
       }
 
       if (
         Number.isFinite(tileJSON.maxzoom) &&
-        (!Number.isFinite(maxZoom) || tileJSON.maxzoom < maxZoom)
+        (!Number.isFinite(props.maxZoom) || tileJSON.maxzoom < props.maxZoom)
       ) {
-        maxZoom = tileJSON.maxzoom;
+        opts.maxZoom = tileJSON.maxzoom;
       }
     }
-
-    this.setState({data, tileJSON, minZoom, maxZoom});
```

Comment:
Can we remove this now?

https://github.com/visgl/deck.gl/blob/5338f9abbcb253c44fcc3464676591255724431a/modules/geo-layers/src/tile-layer/tile-layer.js#L85-L86

Reply:
As we've redefined `_getTilesetOptions`, maxZoom and minZoom are directly returned no need to be in the state. 

I'm thinking now that this two lines https://github.com/visgl/deck.gl/blob/5338f9abbcb253c44fcc3464676591255724431a/modules/geo-layers/src/tile-layer/tile-layer.js#L85-L86 can be removed now.